### DISCUSSION TITLE
Support AWS XRay Id format for Grafana internal traces with Open Telemetry

### DISCUSSION THREAD

**Discussion by faridnsh** [grafana/grafana#45638] [Platform]
Hi,

I noticed recently support for Open Telemetry tracing protocol was added for Grafana internal traces. Using AWS Open Telemetry collector I would like to send these traces to AWS XRay, however AWS XRay requires the trace IDs to be in a different format.

I was hoping we could add an option to Grafana config to allow Grafana to use AWS XRay ID format for internal traces. When the option is enabled something we just need to something `trace.WithIDGenerator(xray.NewIDGenerator())`  to [here](https://github.com/grafana/grafana/blob/v8.4.0/pkg/infra/tracing/opentelemetry_tracing.go#L83) when enabled.

I would be very happy to make a pull request for this.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
